### PR TITLE
Change spongycastle to bouncycastle

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
Addresses https://github.com/dimagi/commcare-android/issues/2218

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
The build fails right now with below error: 
```
Duplicate class org.bouncycastle.crypto.BlockCipher found in modules jetified-android-zebra-interface-1.1-runtime.jar (:android-zebra-interface-1.1:) and jetified-bcprov-jdk15on-1.65.jar (org.bouncycastle:bcprov-jdk15on:1.65)
```
`android-zebra-interface-1.1.aar` library, that we use, has a dependency on `snmp6_1.jar` in the libs folder which contains these bouncycastle code. It seems like it's not possible to ignore the package that's in the libs/ folder of an aar.

File: app/build.gradle [hunk 1]
```diff
@@ -58,16 +58,15 @@ dependencies {
...
     implementation 'com.simprints:LibSimprints:1.0.12'
     implementation 'androidx.multidex:multidex:2.0.1'
     implementation 'androidx.cardview:cardview:1.0.0'
     implementation 'androidx.recyclerview:recyclerview:1.1.0'
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     implementation 'androidx.legacy:legacy-preference-v14:1.0.0'
     implementation 'androidx.gridlayout:gridlayout:1.0.0'
-    implementation 'com.madgag.spongycastle:core:1.54.0.0'
-    implementation 'com.madgag.spongycastle:prov:1.54.0.0'
+    implementation 'org.bouncycastle:bcprov-jdk15on:1.65'
```
Comment (ShivamPokhriyal) [inline app/build.gradle:69]: The latest version of bouncycastle `1.66` is currently giving a build error: `Failed to transform org.bouncycastle/bcprov-jdk15on/1.66 using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 59.` So using `1.65`.

**Comment by ShivamPokhriyal** [comment]
> The build fails right now with below error:
> 
> ```
> Duplicate class org.bouncycastle.crypto.BlockCipher found in modules jetified-android-zebra-interface-1.1-runtime.jar (:android-zebra-interface-1.1:) and jetified-bcprov-jdk15on-1.65.jar (org.bouncycastle:bcprov-jdk15on:1.65)
> ```
> 
> `android-zebra-interface-1.1.aar` library, that we use, has a dependency on `snmp6_1.jar` in the libs folder which contains these bouncycastle code. It seems like it's not possible to ignore the package that's in the libs/ folder of an aar.

I had to remove `org.bouncycastle.crypto` package from `snmp6_1.jar` to fix this error https://github.com/dimagi/zebra-print-android/pull/5
FYI @ctsims

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
[0% (0.0%) vs master 0%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/461801/)

**Comment by ctsims** [comment]
Good news! After my first test printer failed me I was able to find the second one back at the office.

image

Test looks good. Merge away.

**Comment by ShivamPokhriyal** [comment]
> Test looks good. Merge away.

Awesome. Thanks a lot for testing it out!

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Pull request merged**